File: sessions/src/main/java/com/planorama/controller/request/SessionRequest.java
package com.planorama.controller.request;

import com.planorama.domain.Captions;
import com.planorama.domain.User;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.DayOfWeek;
import java.time.LocalDateTime;
import java.util.List;

public record SessionRequest(@NotBlank String scheduleId, Captions captions, User patient, @NotBlank String title,
                             @NotNull LocalDateTime startTime,
                             @NotNull LocalDateTime endTime, String description, List<DayOfWeek> daysOfWeeks) {
}

File: planorama-frontend/src/app/components/modal/domain/session_request.ts
import { Captions } from './captions';

export class SessionRequest {
  public scheduleId!: string;
  public caption: Captions = new Captions();
  public patient!: any;
  public title!: string;
  public startTime!: string;
  public endTime!: string;
  public description!: string;
  public daysOfWeeks: DayOfWeek[] = [];
  public startDate!: Date;
  public endDate!: Date;
}

enum DayOfWeek {
  MONDAY,
  TUESDAY,
  WEDNESDAY,
  THURSDAY,
  FRIDAY,
  SATURDAY,
  SUNDAY,
}
